cmake/mcuboot.cmake

```diff
@@ -96,9 +96,21 @@ function(zephyr_mcuboot_tasks)
     set(imgtool_extra --key "${keyfile}" ${imgtool_extra})
   endif()
 
-  # Use overwrite-only instead of swap upgrades.
   if(CONFIG_MCUBOOT_IMGTOOL_OVERWRITE_ONLY)
+    # Use overwrite-only instead of swap upgrades.
     set(imgtool_extra --overwrite-only --align 1 ${imgtool_extra})
+  elseif(CONFIG_MCUBOOT_BOOTLOADER_MODE_RAM_LOAD)
+    # RAM load requires setting the location of where to load the image to
+    dt_chosen(chosen_ram PROPERTY "zephyr,sram")
```

Comment:
There is a requirement that MCUBoot execute using a different SRAM region than `zephyr,sram` is set to in the customer application using this approach, right? IE if the application is to be loaded to 0x2000_0000-0x2010_0000 MCUBoot must use a different region of SRAM- likely meaning that MCUBoot (or the customer application) will need to change the properties of the `zephyr,sram` chosen node to limit what RAM can be used. Otherwise the ramload process would of course overwrite the RAM MCUBoot was using. Is there somewhere where we can specify that (perhaps in the Kconfig help string)? My core concern here is that this is a somewhat tricky process to set up if you are used to XIP programs, so documenting this thoroughly would be helpful

Reply:
I've not looked at this in a while so would need to go download it and look through to understand but I think it's just the start address of the RAM in the application, so if your RAM goes from 0x20000000 to 0x2010000 then you use the full range in the board by default, for MCUboot you limit the amount of RAM it uses so say up to 0x2000f000 then the application starts at 0x2000f000 and ends at 0x2010000, therefore MCUboot has it's own dedicated RAM, the image gets loaded to 0x2000f000 and any ram after that is used for objects/heap/stack/etc.

Reply:
Yeah, I think we are in agreement here- basically (if I remember right from the last time I used the ramload feature) the application would need to be built with an SRAM region that does not overlap with the MCUBoot SRAM region. Both regions can be in the same physical SRAM device, but they must use different parts of it.

All I was saying here (and it is just a suggestion) is that we document this well, to make it as easy as possible for developers looking to enable ramload on their board.